packages/utilities/tests/throttle.test.ts

```diff
@@ -0,0 +1,26 @@
+import { throttle, sleep } from '../src';
+
+describe('throttle', () => {
+	test('GIVEN number callback THEN returns the same output until the delay', async () => {
+		const callback = vi.fn((num: number) => num);
+
+		const throttleFunc = throttle(callback, 100);
+		const now = Date.now();
+		expect(throttleFunc(now)).toEqual(now);
+		expect(throttleFunc(100)).toEqual(now);
+		expect(callback).toHaveBeenCalledOnce();
+		await sleep(100);
+		expect(throttleFunc(200)).toEqual(200);
```

Comment:
This is flaky, please let the throttle wait be at least half the time of the sleep to leave +50ms variance.